src/js/actions/index.js

```diff
@@ -0,0 +1,203 @@
+import types from '../constants/action-types';
+
+// Call action dispatch to set the filterValue string when filtering the dog list
+export function setDogFilter(filterValue) {
+    return {
+        type: types.SET_DOG_FILTER,
+        filterValue
+    };
+}
+
+// Looks for a key in localStorage and calls an action to set it in store
+// If key don't exist calls an action to fetch it
+export function getDogs() {
+    const dogsFromLocalStorage = getFromLocalStorage('dogList');
+    if (dogsFromLocalStorage) {
+        return setDogList(dogsFromLocalStorage);
+    }
+    return fetchDogs();
+}
+
+// Looks for a key in localStorage and returns its value
+// @param {string} key - key to look for
+// @param {bool} parse - should the retrieved value be parsed
+// Returns: whatever was found in local storage
+export function getFromLocalStorage(key, parse = true) {
+    const dogs = localStorage.getItem(key)
+        ? parse
+            ? JSON.parse(localStorage.getItem(key))
+            : localStorage.getItem(key)
+        : false;
+    return dogs;
+}
+
+// Fetch dog list from API and format it to an array of objects,
+// calls the actions to set the list or error corresponding to the fetch response
+// @params {string} url - url to API
+export function fetchDogs(url = 'https://dog.ceo/api/breeds/list/all') {
+    return dispatch =>
+        fetch(url)
+            .then(r => {
+                if (r.status >= 400) {
+                    throw new Error('Unable to fetch dog list.');
+                }
+                return r.json();
+            })
+            .then(r => {
+                const dogList = Object.keys(r.message).map(dog => ({
+                    name: dog,
+                    subBreed: r.message[dog],
+                    imgURL: '',
+                    imgError: ''
+                }));
+                dispatch(setDogList(dogList));
+            })
+            .catch(error => dispatch(setDogListError(error.toString())));
+}
+
+// Fetch dog image from API calling the actions to set dog image url
+// or error corresponding to the fetch response
+// @param {string} dogBreed - breed of the dog to fetch image
+// @param {array} dogList - array of objects of dogs
+export function fetchDogImg(dogBreed, dogList) {
+    const imgEndpoint = `https://dog.ceo/api/breed/${dogBreed}/images/random`;
+    const dog = dogList.find(dog => dog.name === dogBreed);
+    return dispatch => {
+        return fetch(imgEndpoint)
```

Comment:
Remove the return statement, remember that arrow functions have implicit return.

Reply:
Done.